patches/chromium/feat_add_support_for_missing_dialog_features_to_shell_dialogs.patch

```diff
@@ -0,0 +1,236 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Shelley Vohr <shelley.vohr@gmail.com>
+Date: Sun, 5 May 2024 09:17:17 +0000
+Subject: feat: add support for missing dialog features to //shell_dialogs
+
+This CL adds support for the following features to //shell_dialogs:
+* buttonLabel - Custom label for the confirmation button.
+* showHiddenFiles - Show hidden files in dialog.
+* showOverwriteConfirmation - Whether the user will be presented a confirmation dialog if the user types a file name that already exists.
+
+This may be partially upstreamed to Chromium in the future.
+
+diff --git a/ui/gtk/select_file_dialog_linux_gtk.cc b/ui/gtk/select_file_dialog_linux_gtk.cc
+index 698f97b23f851c02af037880585c82d4494da63f..2455d19fc4c470b640825922c04bfa71adf9bea7 100644
+--- a/ui/gtk/select_file_dialog_linux_gtk.cc
++++ b/ui/gtk/select_file_dialog_linux_gtk.cc
+@@ -243,6 +243,10 @@ void SelectFileDialogLinuxGtk::SelectFileImpl(
+ 
+   std::string title_string = base::UTF16ToUTF8(title);
+ 
++  ExtraSettings extra_settings = std::make_tuple("", true, false);
++  if (params)
++    extra_settings = *(reinterpret_cast<ExtraSettings*>(params));
```

Comment:
```suggestion
+    extra_settings = *static_cast<ExtraSettings*>(params);
```

`void*` is a special case, use `static_cast` there instead of `reinterpret_cast`